### ISSUE TITLE
Peer/Dev Dependency Resolution Error

### ISSUE THREAD

**Issue by harrysolovay** [lerna/lerna#2822]
I have two packages in my monorepo, one of which dev-depends and peer-depends on the other. For this example, I've swapped out the real names with `a` and `b`.

`a/package.json`

```json
{
  "name": "b",
  "version": "0.1.0",
  "main": "dist/cjs/index.js",
  "types": "dist/cjs/index.d.ts",
  "license": "Apache-2.0"
}
```

`b/package.json`

```json
{
  "name": "b",
  "version": "0.1.0",
  "main": "dist/cjs/index.js",
  "types": "dist/cjs/index.d.ts",
  "license": "Apache-2.0",
  "devDependencies": {
    "a": "^0.1.0"
  },
  "peerDependencies": {
    "a": "^0.1.0"
  }
}
```

Upon bootstrapping I encounter the following error:

```
lerna ERR! npm install exited 1 in 'b'
lerna ERR! npm install stderr:
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: a@undefined
npm WARN node_modules/a
npm WARN   peer a@"^0.1.0" from the root project
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer a@"^0.1.0" from the root project
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/a - Not found
npm ERR! 404 
npm ERR! 404  'a@^0.1.0' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.
```

It appears that––for some reason––the version of `a` is `undefined`, hence the package is not resolved. Because the package does not exist on NPM, the installation fails altogether.

Any help / advice would be greatly appreciated! Thank you!

<details><summary>lerna.json</summary><p>

```json
{
  "$schema": "https://json.schemastore.org/lerna",
  "npmClient": "npm",
  "packages": ["packages/*"],
  "version": "0.1.0"
}
```
</p></details>

## Your Environment

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 3.22.1 |
| `npm --version`  | 7.5.2 |
| `node --version` | 14.15.5 |

| OS | Version |
| --- | --- |
| macOS Catalina | 10.15.7 |

**Comment by jl-rojas** [comment]
Maybe the error is: the "name" field in `a/package.json` is wrong. You typed "name": "b" in both JSON files

**Comment by harrysolovay** [comment]
Just updated the description. & no––sadly that's just for the sake of this issue. I believe this is an issue regarding Lerna & NPM 7.

**Comment by adityak74** [comment]
I am also facing the same issue. Here is a snapshot of the error. This is not just specific to this dependency. 

<img width="830" alt="Screen Shot 2021-02-28 at 2 56 33 AM" src="https://user-images.githubusercontent.com/5231288/109412957-92230a00-7970-11eb-95bd-c3d1d6fcabd9.png">

**Comment by adityak74** [comment]
I got this working by clearing my node and npm and installing the latest LTS version from nodejs org.

**Comment by trusktr** [comment]
Indeed, if you run `npm install --legacy-peer-deps` and your `install` script (triggered by `npm install`) runs `lerna bootstrap`, then that `--legacy-peer-deps` option will be passed along to the lerna's install process.

**Comment by iamtomcat** [comment]
If you're using npm 7+ now, you just have to add   `"useWorkspaces": true` as npm has similar behaviour to yarn now.

**Comment by nicojs** [comment]
> If you're using npm 7+ now, you just have to add `"useWorkspaces": true`

If I wanted to use workspaces, why even use lerna?

I don't like the way npm and yarn symlink all my packages in the root node_modules. This doesn't scale for large monorepos where you want to make sure only your direct dependencies are importable.

I think `lerna bootstrap` should temporary replace the peerDependencies during the bootstrapping, similarly as they now replace the devDependencies and dependencies.

**Comment by ryanirilli** [comment]
@nicojs 
> If I wanted to use workspaces, why even use lerna?
I use lerna for publishing only, and avoid lerna boostrap altogether by using npm workspaces

**Comment by rennerg** [comment]
Is anyone looking into this issue? This is blocking us as well.

**Comment by JamesHenry** [comment]
Hi Folks,

This issue related to the legacy package management capabilities of Lerna. The world in which Lerna had to manage package installation and linking concerns because of gaps in the ecosystem no longer exists, we are blessed to have many great options now, not just `npm` (which has also massively improved since the early days).

In June 2023, in Lerna `7.0.0`, we removed the `lerna bootstrap`, `lerna add` and `lerna link` commands in Lerna by default. Over two years later, in September 2025 (Lerna `9.0.0`), the commands were finally fully removed.

We provided some guidance at the time on how to migrate existing codebases to use native package manager features here: https://lerna.js.org/docs/legacy-package-management

I'll close this one as no action is applicable, but please always feel free to open a fresh issue giving as much context as possible when using the latest and greatest and we'll take a look. I'm making sure we close older issues like this one to free up bandwidth to make sure we tackle what matters most to users in 2026.

Many thanks for your contributions to Lerna!
